Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm

```diff
@@ -321,6 +327,19 @@ - (void)hideBanner
 #endif
 }
 
+#if ENABLE(FULLSCREEN_DISMISSAL_GESTURES)
+- (void)_bannerDismissalRecognized:(NSNotification*)notification
+{
+    auto remainingDelay = _bannerMinimumHideDelayTime - MonotonicTime::now();
+    if (remainingDelay <= 0_s) {
+        [self hideBanner];
+        return;
+    }
+
+    [self performSelector:@selector(hideBanner) withObject:nil afterDelay:remainingDelay.seconds()];
```

Comment:
Do we also need a `cancelPreviousPerformRequestsWithTarget` when the banner is displayed, so that we don't run into a scenario where dismiss immediately when the banner is brought up again?

Reply:
Both `-hideBanner` and `-showBanner` will `cancelPrevious...`. So I don't think there's a scenario where the tap-to-dismiss behavior will cause the banner to immediately hide after showing.

Reply:
Thanks!